File: backend/backend_JS/src/services/ProfileService.ts
import Profile, { IProfile } from '../models/Profile';

class ProfileService {
  /**
   * Get a profile by ID
   */
  async getProfileById(profileId: string): Promise<IProfile | null> {
    return await Profile.findOne({ profileId: profileId });
  }

  /**
   * Get all profiles
   */
  async getAllProfiles(): Promise<IProfile[]> {
    return await Profile.find();
  }

  /**
   * Create a new profile
   */
  async createProfile(profileData: Partial<IProfile>): Promise<IProfile> {
    if (!profileData.profileId) {
      throw new Error('profileId is required to create a profile.');
    }
    const profile = new Profile(profileData);
    return await profile.save();
  }

  /**
   * Update an existing profile
   */
  async updateProfile(profileId: string, profileData: Partial<IProfile>): Promise<IProfile | null> {
    return await Profile.findOneAndUpdate({ profileId: profileId }, profileData, { new: true });
  }

  /**
   * Delete a profile
   */
  async deleteProfile(profileId: string): Promise<boolean> {
    const result = await Profile.deleteOne({ profileId: profileId });
    return result.deletedCount > 0;
  }

  /**
   * Check if any profiles exist in the database
   */
  async profilesExist(): Promise<boolean> {
    const count = await Profile.countDocuments();
    return count > 0;
  }
}

export default new ProfileService(); 

File: backend/backend_JAVA/src/main/java/backend/services/ProfileService.java
package backend.services;

import backend.exceptions.ResourceNotFoundException;
import backend.models.Profile;
import backend.repositories.ProfileRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProfileService {
    
    private final ProfileRepository profileRepository;
    
    public ProfileService(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }
    
    public Profile getProfileById(String profileId) {
        return profileRepository.findByProfileId(profileId)
                .orElseThrow(() -> new ResourceNotFoundException("Profile", "profileId", profileId));
    }
    
    public List<Profile> getAllProfiles() {
        return profileRepository.findAll();
    }
    
    public Profile createProfile(Profile profile) {
        if (profile.getProfileId() == null) {
            throw new IllegalArgumentException("profileId is required to create a profile");
        }
        
        return profileRepository.save(profile);
    }
    
    public Profile updateProfile(String profileId, Profile profileDetails) {
        Profile profile = profileRepository.findByProfileId(profileId)
                .orElseThrow(() -> new ResourceNotFoundException("Profile", "profileId", profileId));
        
        // Update profile fields
        profile.setBusinessName(profileDetails.getBusinessName());
        profile.setLogo(profileDetails.getLogo());
        profile.setCoverImage(profileDetails.getCoverImage());
        profile.setIsPro(profileDetails.getIsPro());
        profile.setIsVerified(profileDetails.getIsVerified());
        profile.setRevenue(profileDetails.getRevenue());
        profile.setEmployeeCount(profileDetails.getEmployeeCount());
        profile.setBusinessOverview(profileDetails.getBusinessOverview());
        profile.setBusinessType(profileDetails.getBusinessType());
        profile.setOrigin(profileDetails.getOrigin());
        profile.setEstablished(profileDetails.getEstablished());
        profile.setExportVolume(profileDetails.getExportVolume());
        profile.setWebsite(profileDetails.getWebsite());
        profile.setAddress(profileDetails.getAddress());
        profile.setMobile(profileDetails.getMobile());
        profile.setOwner(profileDetails.getOwner());
        profile.setVerifications(profileDetails.getVerifications());
        profile.setCertifications(profileDetails.getCertifications());
        profile.setImportExport(profileDetails.getImportExport());
        
        return profileRepository.save(profile);
    }
    
    public boolean deleteProfile(String profileId) {
        if (profileRepository.findByProfileId(profileId).isPresent()) {
            profileRepository.deleteByProfileId(profileId);
            return true;
        }
        return false;
    }
    
    public boolean profilesExist() {
        return profileRepository.count() > 0;
    }
} 
